webservices/ruby/style_guide.md

```diff
@@ -0,0 +1,492 @@
+        true
+      end
+    end
+
+  # also good
+    def best_pizza?
+      if tasty_pizza && !small_pizza?
+        true
+      end
+    end
+  ```
+
+# Blocks
+### Prefer &:method_name for simple method calls.
+  ```ruby
+    # bad
+    def delicious_pizza?
+      toppings.detect{ |topping| topping.delicious }
+    end
+
+    # good
+    def delicious_pizza?
+      toppings.detect(&:delicious)
+    end
+  ```
+
+### Use {...} for single-line blocks. Use do..end for multi-line blocks.
+  ```ruby
+    # bad
+      def choose_best_vacations(places)
+        places.select { |place|
+          if place.super_fun?
+            place
+          end
+        }
+      end
+
+    # also bad
+    def choose_best_vacations(places)
+      places.select do |place| place if place.super_fun? end
+    end
+
+    # good
+    def choose_best_vacations(places)
+      places.select do |place|
+        if place.super_fun?
+          place
+        end
+      end
+    end
+
+    # also good
+    def choose_best_vacations(places)
+      places.select { |place| place if place.super_fun? }
+    end
+  ```
+
+### Prefer detect, select, map, and reduce for iteration.
+  ```ruby
+    # bad
+    def find_elephants(animals)
+      elephants = []
+      animals.each do |animal|
+        if animal.elephant
+          elephants << animal
+        end
+      end
+    end
+
+    # good
+    def find_elephants(animals)
+      animals.select do |animal|
+        if animal.elephant?
+          animal
+      end
+    end
+  ```
+
+# Classes & Methods
+### Use ? suffix for predicate methods, and avoid prefixing auxiliary verbs.
+  ```ruby
+    # bad
+    def is_big
+      dinosaur.size == "big"
+    end
+
+    # also bad
+    def is_big?
+      dinosaur.size == "big"
+    end
+
+    # good
+    def big?
+      dinosaur.size == "big"
+    end
+  ```
+
+### Use def with parentheses when there are arguments.
+  ```ruby
+    # bad
+    def find_elephants animals
+      animals.select do |animal|
+        if animal.elephant?
+          animal
+      end
+    end
+
+    # good
+    def find_elephants(animals)
+      animals.select do |animal|
+        if animal.elephant?
+          animal
+      end
+    end
+  ```
+
+### Don't use spaces after required keyword arguments.
+  ```ruby
+    # bad
+    def sum(operand_1: , operand_2: , operand_3:)
+      operand_1 + operand_2 + operand_3
+    end
+
+    # good
+    def sum(operand_1:, operand_2:, operand_3:)
+      operand_1 + operand_2 + operand_3
+    end
+  ```
+
+### Avoid bang method names
+  - Bang (!) usually denotes a "dangerous" method, meaning that it will modify the object it's called on.
+  ```ruby
+    # bad
+    class Bus
+      def number_of_wheels!(bus_params)
+        self.number_of_wheels = bus_params[:number_of_wheels]
+        self.save
+      end
+    end
+
+    # good
+    class Bus
+      def update_number_of_wheels(bus_params)
+        self.number_of_wheels = bus_params[:number_of_wheels]
+        self.save
```

Comment:
could there also be another "good" example still keeping the method def as `update_number_of_wheels!` but using `self.save!` to match the expectations?